src/api/app/controllers/webui/cloud/upload_jobs_controller.rb

```diff
@@ -46,12 +48,19 @@ def set_breadcrump
         @crumb_list = ['Cloud Upload']
       end
 
+      def validate_uploadable
+        return if ::Cloud::UploadJob.uploadable_file?(params[:filename]) &&
```

Comment:
How about ::Cloud::UploadJob.uploadable?(params)?

Reply:
Looks like you have it already^^ -> https://github.com/openSUSE/open-build-service/pull/4331/files#diff-b6336beca5d02eb47a478e4ef107cf55R90

Reply:
I have it for the view, yes! Point is, I also want to have it separate for the model validation to make the error message more verbose ... :/

Reply:
Got it, but we could still have an uploadable method

Reply:
> Point is, I also want to have it separate for the model validation to make the error message more verbose

Why does having the code seperate between helper & model mean that you can make the error messages more verbose? Its the exact same code isn't it?

```ruby
   ::Cloud::UploadJob.uploadable_file?(filename) &&
     ::Cloud::UploadJob.uploadable_architecture?(architecture)
```

Reply:
Seems like we should have these validations done in the model not the controller? Since you already call `if @upload_job.valid?` in the create action?

Reply:
Yeah but this is for the :new action because otherwise you could go to ``cloud/new/project/package/arch/repo/SOMERANDOMFILE`` and I want to avoid that you can enter the :new page with some random file and validate that is actually a cloud image.

Reply:
Actually nevermind that last comment.. It seems my brain is taking a while to get working this morning :p

What about doing something like this instead, to remove duplicated logic and also move logic from the controller to model:
```ruby
      def new
        xml_object = OpenStruct.new(params.slice(:project, :package, :repository, :arch, :filename))
        @upload_job = ::Cloud::Backend::UploadJob.new(xml_object: xml_object)
        unless @upload_job.uploadable?
          flash[:error] = "File '#{params[:filename]}' with architecture '#{params[:arch]}' is not a valid cloud image."
          redirect_to cloud_upload_index_path
        end
```

Also why do we need to check that its uploadable on the new page?